app/src/main/java/com/github/multimatum_team/multimatum/util/PDFUtil.kt

```diff
@@ -25,9 +22,28 @@ object PDFUtil {
     /**
      * Retrieve name of a pdf from it's Uri
      */
-    fun getFileNameFromUri(pdfData: Uri): String {
-        val path: String = pdfData.toString()
-        val lastSlashIndex = path.lastIndexOf("/")
-        return path.substring(lastSlashIndex + 1, path.length)
+    fun getFileNameFromUri(pdfData: Uri, context: Context): String {
+        val cursor: Cursor? = context.contentResolver.query(pdfData, null, null, null, null)
+        var fileName = ""
+        if (cursor != null) {
+            if (cursor.count <= 0) {
+                cursor.close()
+                throw IllegalArgumentException("Can't obtain file name, cursor is empty")
+            }
+            cursor.moveToFirst()
+            fileName =
+                cursor.getString(cursor.getColumnIndexOrThrow(OpenableColumns.DISPLAY_NAME))
+            cursor.close()
+        } else {
+            val pathStr = pdfData.toString()
+            fileName = pathStr.substring(pathStr.lastIndexOf('/') + 1).drop(16)
+        }
+        return fileName
+    }
+
+    fun addRdmCharToStr(input: String, numOfChar: Int): String {
```

Comment:
This is not a good way to generate unique identifiers as it can generate the same ID twice (even if unlikely).
Generating unique IDs is not simple to do, you could rely on the `UUID` class in the java standard library.
<https://developer.android.com/reference/java/util/UUID>